From 29e98c05620520a2e7c7505ee5eafce97577393d Mon Sep 17 00:00:00 2001
From: Viswanath Kraleti <viswanath.kraleti@oss.qualcomm.com>
Date: Thu, 24 Sep 2026 08:02:09 +0530
Subject: [PATCH] linux-firmware: backport patch to symlink shikra wifi
 firmware

On Shikra platform the WiFi firmware is loaded by the tqftpserv
from the same location as the rest of the DSP firmware. This link
was missing in the previous commits that brought it in WiFi firmware.

Bring the upstream commit 837ae0e5becc ("qcom/shikra: link WiFi
firmware from the ath10k subdir") that is pointing from the expected
location to the actual WiFi firmware file as a patch file.

Signed-off-by: Viswanath Kraleti <viswanath.kraleti@oss.qualcomm.com>
Upstream-Status: Backport [https://gitlab.com/kernel-firmware/linux-firmware/-/commit/837ae0e5beccc2f1e974afba56c22c3cf981fc4e]
---
 ...-WiFi-firmware-from-the-ath10k-subdi.patch | 28 +++++++++++++++++++
 .../linux-firmware/linux-firmware_20260910.bb |  3 +-
 2 files changed, 30 insertions(+), 1 deletion(-)
 create mode 100644 recipes-kernel/linux-firmware/linux-firmware/0001-qcom-shikra-link-WiFi-firmware-from-the-ath10k-subdi.patch

diff --git a/recipes-kernel/linux-firmware/linux-firmware/0001-qcom-shikra-link-WiFi-firmware-from-the-ath10k-subdi.patch b/recipes-kernel/linux-firmware/linux-firmware/0001-qcom-shikra-link-WiFi-firmware-from-the-ath10k-subdi.patch
new file mode 100644
index 0000000..010121d
--- /dev/null
+++ b/recipes-kernel/linux-firmware/linux-firmware/0001-qcom-shikra-link-WiFi-firmware-from-the-ath10k-subdi.patch
@@ -0,0 +1,28 @@
+From 837ae0e5beccc2f1e974afba56c22c3cf981fc4e Mon Sep 17 00:00:00 2001
+From: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
+Date: Fri, 18 Sep 2026 17:52:15 +0300
+Subject: [PATCH] qcom/shikra: link WiFi firmware from the ath10k subdir
+
+On Qualcomm Shikra platform the WiFi firmware is loaded by the tqftpserv
+from the same location as the rest of the DSP firmware. Add symlink,
+pointing from the expected location to the actual WiFi firmware file.
+
+Fixes: f64d7db7d264 ("ath10k: WCN3990: hw1.0: add shikra firmware files")
+Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
+Upstream-Status: Backport [https://gitlab.com/kernel-firmware/linux-firmware/-/commit/837ae0e5beccc2f1e974afba56c22c3cf981fc4e]
+---
+ WHENCE | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/WHENCE b/WHENCE
+index a712978b..053640a0 100644
+--- a/WHENCE
++++ b/WHENCE
+@@ -3825,6 +3825,7 @@ Version: WLAN.HL.2.0-01387-QCAHLSWMTPLZ-1
+ File: ath10k/WCN3990/hw1.0/notice.txt_wlanmdsp
+ File: ath10k/WCN3990/hw1.0/shikra/firmware-5.bin
+ File: ath10k/WCN3990/hw1.0/shikra/wlanmdsp.mbn
++Link: qcom/shikra/wlanmdsp.mbn -> ../../ath10k/WCN3990/hw1.0/shikra/wlanmdsp.mbn
+ Version: WLAN.HL.3.3.7.c5-00093.2-QCAHLSWMTPL-1
+ File: ath10k/WCN3990/hw1.0/shikra/notice.txt_wlanmdsp
+ 
diff --git a/recipes-kernel/linux-firmware/linux-firmware_20260910.bb b/recipes-kernel/linux-firmware/linux-firmware_20260910.bb
index abaacda..1132f44 100644
--- a/recipes-kernel/linux-firmware/linux-firmware_20260910.bb
+++ b/recipes-kernel/linux-firmware/linux-firmware_20260910.bb
@@ -277,7 +277,7 @@ LIC_FILES_CHKSUM = "file://LICENSES/LICENCE.Abilis;md5=b5ee3f410780e56711ad48ead
                     "
 # WHENCE checksum is defined separately to ease overriding it if
 # class-devupstream is selected.
-WHENCE_CHKSUM  = "47dbfc1b4d2d5a13c2e990ae38c3ac2b"
+WHENCE_CHKSUM  = "0468f24519fa953ad96a5c89e9f22812"
 
 # These are not common licenses, set NO_GENERIC_LICENSE for them
 # so that the license files will be copied from fetched source
@@ -412,6 +412,7 @@ PATCHTOOL:qcom = "git"
 SRC_URI = "\
   ${KERNELORG_MIRROR}/linux/kernel/firmware/${BPN}-${PV}.tar.xz \
   file://0001-ath10k-WCN3990-hw1.0-add-shikra-firmware-files.patch \
+  file://0001-qcom-shikra-link-WiFi-firmware-from-the-ath10k-subdi.patch \
 "
 
 BBCLASSEXTEND = "devupstream:target"
